tools/mpremote/tests/test_filesystem.sh

```diff
@@ -170,3 +170,20 @@ EOF
 $MPREMOTE resume cp -r "${TMP}/package" :
 $MPREMOTE resume ls : :package :package/subpackage
 $MPREMOTE resume exec "import package; package.x(); package.y()"
+
+# Test rm -r functionality
+echo -----
+$MPREMOTE resume mkdir :testdir
+$MPREMOTE resume cp -r "${TMP}/package" :testdir/package
+$MPREMOTE resume ls :testdir
+$MPREMOTE resume rm -r :testdir/package
+$MPREMOTE resume ls :testdir
+
+# Test rm -r on non-existent path
+echo -----
+$MPREMOTE resume rm -r :nonexistent || echo "expect error"
+
+# Test rm -r on read-only filesystem
+echo -----
+$MPREMOTE resume exec "import os; os.mount(os.VfsFat(RAMBlockDev(512, 50)), '/readonly', readonly=True)"
+$MPREMOTE resume rm -r :readonly || echo "expect error"
```

Comment:
Found another issue when running the test suite: the expected outputs haven't been added to `test_filesystem.sh.exp`!

Reply:
Ah , that's indeed a bummer when you have tests that can't run on some platforms. 
All see if I can generate these in WSL2 as well.

Reply:
After inspecting the serial traces, I think this is the same issue I've had before with mpremote basically being just completely unusably broken. My rp2040 is just constantly soft-rebooting after just about every command; it's perhaps more impressive that the test even gets as far as it does before failing.

Reply:
That's not the issue here: the test harness includes a step that compares the output of `test_filesystem.sh` to the contents of `test_filesystem.sh.exp`.

This PR does not include _anything_ that would update `test_filesystem.sh.exp` with the additional lines of output that the additional lines of test code would generate.

Did you forget to stage and commit this file?

Reply:
I missed clear docs on how this could be done on Windows.